### ISSUE TITLE
If the async task throw undefined, the error collection will fail

### ISSUE THREAD

**Issue by wzhkobe2408** [supercharge/promise-pool#18]
The code below excute async task that will reject with nothing.

Therefore the `startProcessing` function will fail at `new PromisePoolError(error.message, item)`.

I know this is not common in real world, but it's it nessary to improve robustness ?

```javascript
async function main () {
  const { results, errors } = await PromisePool
    .for(Users)
    .withConcurrency(3)
    .process(async (user) => {
      await new Promise((resolve, reject) => setTimeout(reject, 1000))

      return user
    })

  console.log('results', results)
  console.log('errors', errors)
}

main()
```

**Comment by marcuspoehls** [comment]
@wzhkobe2408 good catch! Thank you for reporting this. 

I'l need to run your sample code to see the result. Do you want to send a pull request improving the handling?

**Comment by wzhkobe2408** [comment]
@marcuspoehls  Yes I do. I will lauch a pr later 😊

**Comment by wzhkobe2408** [comment]
@marcuspoehls    pr link: https://github.com/supercharge/promise-pool/pull/19

**Comment by marcuspoehls** [comment]
I just tagged and published version `1.4.0` to NPM. Thank you for your work!
